webapp/store/models.py

```diff
@@ -0,0 +1,172 @@
+        'owner': entity_data.get('owner', {}).get('User'),
+        'provides': meta['charm-metadata'].get('Provides'),
+        'requires': meta['charm-metadata'].get('Requires'),
+        'resources': _extract_resources(ref, meta.get('resources')),
+        'readme': _render_markdown(cs.entity_readme_content(charm_or_bundle_id)),
+        'revision_list': revision_list,
+        'revision_number': ref.revision,
+        'revisions': revisions,
+        'series': meta.get('supported-series', {}).get('SupportedSeries'),
+        'is_charm': 'charm-metadata' in meta
+    }
+
+    return charm_or_bundle
+
+
+def _getEntityFiles(ref, manifest=None):
+    try:
+        files = cs.files(ref, manifest=manifest)
+    except EntityNotFound:
+        files = {}
+    if not files:
+        files = {}
+    try:
+        readme = cs.entity_readme_content(ref)
+    except EntityNotFound:
+        readme = None
+    return {
+        'files': files,
+        'readme': readme,
+    }
+
+
+def _parse_charm_icon(charm_id):
+    try:
+        return cs.charm_icon_url(charm_id)
+    except Exception:
+        return abort(500, "Entity not found {}".format(charm_id))
+
+
+def _expand_revision(revision_id, charm_name):
+    '''From a revision id, create a Dict.
+    @param revision_id A revision id string.
+    @param charm_name The name of the charm
+    '''
+
+    id = revision_id.split('-')[-1]
+    url = charm_name + '/' + id
+    return {
+        'id': int(id),
+        'full_id': revision_id,
+        'url': url}
+
+
+def _extract_resources(ref, data):
+    """Extract data from resources metadata.
+    @param ref The reference of the entity.
+    @param data the resources metadata associated with the entity.
+    @return a dictionary of resource name and an array containing
+            the file extension, the file link of the resource.
+    """
+    if not data:
+        # if resources are not present return an empty dictionary
+        return {}
+
+    try:
+        resources = {}
+        for i in data:
+            resource_url = ""
+            if i['Revision'] >= 0:
+                resource_url = self.cs.resource_url(ref,
+                                                    i['Name'],
+                                                    i['Revision'])
+            resources[i['Name']] = [os.path.splitext(i['Path'])[1],
+                                    resource_url]
+    except (KeyError, TypeError, ValueError):
+        log.info('Unable to read resources for ',
+                    data)
+        return {}
+
+    return resources
+
+
+def _get_bug_url(name, bugs_url):
+    '''Create a link to the bug tracker on Launchpad.
+    @param name the charm name.
+    @return a URL for the bug tracker.
+    '''
+    if bugs_url:
+        return bugs_url
+    return 'https://bugs.launchpad.net/charms/+source/{}'.format(name)
+
+
+def _render_markdown(content):
+    try:
+        html = gfm.markdown(content)
```

Comment:
If all you want to do is "leave the README unparsed", then you need to put this line outside the `try..except` block.